x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_overview_header/engine_overview_header.test.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { mount } from 'enzyme';
+
+import { EngineOverviewHeader } from '../engine_overview_header';
+import { KibanaContext } from '../../..';
+
+describe('EngineOverviewHeader', () => {
+  let enterpriseSearchUrl;
+
+  afterEach(() => {
+    enterpriseSearchUrl = undefined;
+  });
+
+  const render = () => {
+    return mount(
+      <KibanaContext.Provider
+        value={{
+          http: {},
+          enterpriseSearchUrl,
+          setBreadcrumbs: jest.fn(),
+        }}
+      >
+        <EngineOverviewHeader />
+      </KibanaContext.Provider>
+    );
+  };
+
+  describe('when enterpriseSearchUrl is set', () => {
+    let wrapper;
+
+    beforeEach(() => {
+      enterpriseSearchUrl = 'http://localhost:3002';
+      wrapper = render();
+    });
```

Comment:
I'd like to see us refactor everything within context to be a passed param rather than let strings.

As an example:

```
  const render = context => {
    return mount(
      <KibanaContext.Provider
        value={{
          http: {},
          enterpriseSearchUrl: '',
          setBreadcrumbs: jest.fn(),
          ...context,
        }}
      >
        <EngineOverviewHeader />
      </KibanaContext.Provider>
    );
  };


  describe('when enterpriseSearchUrl is set', () => {
    beforeEach(() => {
      wrapper = render({ enterpriseSearchUrl: 'http://localhost:3002' });
    });
  });

  describe('when enterpriseSearchUrl is not set', () => {
    beforeEach(() => {
      wrapper = render({ enterpriseSearchUrl: undefined });
    });
```

Let me know how that feels to you?

Reply:
Final form here: https://github.com/constancecchen/kibana/pull/4/files#r415998493

Reply:
Feels good a7b0acd